### ISSUE TITLE
ls command: ability to group items

### ISSUE THREAD

**Issue by bram85** [bram85/topydo#22]
Introduce a `-g` flag in the `ls` subcommand to group todo items, similar to `-s` to specify the sorting order.

Example:

```
# group by due date
topydo ls -g due
Today
  |  1| Foo due:2015-05-16
Tomorrow
  |  2| Bar due:2015-05-17
  |  3| Baz due:2015-05-17

# group by projects
topydo ls -g project
```

It should support all fields that you can pass to `-s`, with the addition of `project` and `context`. Also, you can only pass one field, whereas `-s` accepts multiple.

Still unsure what the output should look like exactly.

**Comment by inducer** [comment]
One carryover from #112 might be the ability to group by multiple items. (e.g. due dates and priority)

**Comment by bram85** [comment]
Yes indeed, multiple items should be feasible.

**Comment by bram85** [comment]
Started working on this in the [groups branch](https://github.com/bram85/topydo/tree/groups). It's still very rough around the edges, but the principle is working.

**Comment by bram85** [comment]
Released with version 0.11
